Sort icon in tables should indicate if the order applies to the column and the direction

##### **Overview of the feature request**

I think the UX could improve a lot if we include an indicator of which column is used for the order, and which direction is used (ASC or DESC).

I fixed in my personal project with a little change:

From

`<th v-on:click="changeOrder('enabled')"><span v-text="$t('carwashApp.vehicleColor.enabled')">Enabled</span> <font-awesome-icon icon="sort"></font-awesome-icon></th>`

To

```
<th v-on:click="changeOrder('enabled')">
    <span v-text="$t('carwashApp.vehicleColor.enabled')">Enabled</span>
    <i class="fas" v-bind:class="propOrder === 'enabled' ? (reverse ? 'fa-sort-down' : 'fa-sort-up') : 'fa-sort'"></i>
</th>
```

This way we have 3 indicators:

No sort

![Screenshot from 2019-09-29 15-40-42](https://user-images.githubusercontent.com/11655127/65839085-92c4f600-e2cf-11e9-97cb-3aa20479cc75.png)

Sort asc

![Screenshot from 2019-09-29 15-40-22](https://user-images.githubusercontent.com/11655127/65839096-bc7e1d00-e2cf-11e9-9d65-445d77c7cc11.png)


And sort desc

![Screenshot from 2019-09-29 15-40-32](https://user-images.githubusercontent.com/11655127/65839095-b5efa580-e2cf-11e9-9c48-260dfbc53175.png)

If you like it, I could prepare a pull request to change the generator, I would use the `font-awesome-icon`, adding the missing icons.

##### **Motivation for or Use Case**

Right now there is no way to check easily which column is selected for the order, this causes confusions and extra clicks to be sure.

##### **Related issues or PR**

I did not found anything open with "sort", "desc", or similar.

-   [ X ] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

I made a pull request, but I think master is not working, in local and in the test, there is some error (before I modify anything)

```
##[section]Starting: GENERATION: jhipster info
==============================================================================
Task         : Bash
Description  : Run a Bash script on macOS, Linux, or Windows
Version      : 3.151.3
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/bash
==============================================================================
Generating script.
Script contents:
$HOME/generator-jhipster/test-integration/scripts/14-jhipster-info.sh
========================== Starting Command Output ===========================
[command]/bin/bash --noprofile --norc /home/vsts/work/_temp/478f4dc5-93e5-4d7b-81e4-71c9387197bc.sh
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'semver'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/vsts/generator-jhipster/cli/jhipster.js:20:16)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
##[error]Bash exited with code '1'.
##[section]Finishing: GENERATION: jhipster info
```

--- a/generators/client/files.js
+++ b/generators/client/files.js
@@ -164,6 +164,8 @@ const vueFiles = {
                 'core/ribbon/ribbon.vue',
                 'core/ribbon/ribbon.component.ts',
                 'shared/date/filters.ts',
+                'shared/sort/jhi-sort-indicator.component.ts',
+                'shared/sort/jhi-sort-indicator.vue',
                 'shared/sort/sorts.ts',
                 'shared/alert/alert.service.ts',
                 'shared/alert/alert.mixin.ts',

--- a/generators/client/templates/vue/src/main/webapp/app/admin/audits/audits.vue.ejs
+++ b/generators/client/templates/vue/src/main/webapp/app/admin/audits/audits.vue.ejs
@@ -26,10 +26,10 @@
             <table class="table table-sm table-striped">
                 <thead>
                 <tr>
-                    <th v-on:click="changeOrder('auditEventDate', 'timestamp')"><span v-text="$t('audits.table.header.date')">Date</span><font-awesome-icon v-if="propOrder === 'auditEventDate'" icon="sort"></font-awesome-icon></th>
-                    <th v-on:click="changeOrder('principal', 'principal')"><span v-text="$t('audits.table.header.principal')">User</span><font-awesome-icon v-if="propOrder === 'principal'" icon="sort"></font-awesome-icon></th>
-                    <th v-on:click="changeOrder('auditEventType', 'type')"><span v-text="$t('audits.table.header.status')">State</span><font-awesome-icon v-if="propOrder === 'auditEventType'" icon="sort"></font-awesome-icon></th>
-                    <th><span v-text="$t('audits.table.header.data')">Extra data</span></th>
+                  <th v-on:click="changeOrder('auditEventDate', 'timestamp')"><span v-text="$t('audits.table.header.date')">Date</span> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'auditEventDate'"></jhi-sort-indicator></th>
+                  <th v-on:click="changeOrder('principal', 'principal')"><span v-text="$t('audits.table.header.principal')">User</span> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'principal'"></jhi-sort-indicator></th>
+                  <th v-on:click="changeOrder('auditEventType', 'type')"><span v-text="$t('audits.table.header.status')">State</span> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'auditEventType'"></jhi-sort-indicator></th>
+                  <th><span v-text="$t('audits.table.header.data')">Extra data</span></th>
                 </tr>
                 </thead>
                 <tbody>

--- a/generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.vue.ejs
+++ b/generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.vue.ejs
@@ -17,20 +17,20 @@
             <table class="table table-striped">
                 <thead>
                 <tr>
-                    <th<% if (databaseType !== 'cassandra') { %> v-on:click="changeOrder('id')"<% } %>><span v-text="$t('global.field.id')">ID</span><% if (databaseType !== 'cassandra') { %> <font-awesome-icon icon="sort"></font-awesome-icon><% } %></th>
-                    <th<% if (databaseType !== 'cassandra') { %> v-on:click="changeOrder('login')"<% } %>><span v-text="$t('userManagement.login')">Login</span><% if (databaseType !== 'cassandra') { %> <font-awesome-icon icon="sort"></font-awesome-icon><% } %></th>
-                    <th<% if (databaseType !== 'cassandra') { %> v-on:click="changeOrder('email')"<% } %>><span v-text="$t('userManagement.email')">Email</span><% if (databaseType !== 'cassandra') { %> <font-awesome-icon icon="sort"></font-awesome-icon><% } %></th>
-                    <th></th>
-                    <%_ if (enableTranslation) { _%>
-                        <th<% if (databaseType !== 'cassandra') { %> v-on:click="changeOrder('langKey')"<% } %>> <span v-text="$t('userManagement.langKey')">Lang Key</span><% if (databaseType !== 'cassandra') { %> <font-awesome-icon icon="sort"></font-awesome-icon><% } %></th>
-                    <%_ } _%>
-                    <th><span v-text="$t('userManagement.profiles')">Profiles</span></th>
-                    <%_ if (databaseType !== 'cassandra') { _%>
-                    <th v-on:click="changeOrder('createdDate')"><span v-text="$t('userManagement.createdDate')">Created Date</span> <font-awesome-icon icon="sort"></font-awesome-icon></th>
-                    <th v-on:click="changeOrder('lastModifiedBy')"><span v-text="$t('userManagement.lastModifiedBy')">Last Modified By</span> <font-awesome-icon icon="sort"></font-awesome-icon></th>
-                    <th id="modified-date-sort" v-on:click="changeOrder('lastModifiedDate')"><span v-text="$t('userManagement.lastModifiedDate')">Last Modified Date</span> <font-awesome-icon icon="sort"></font-awesome-icon></th>
-                    <%_ } _%>
-                    <th></th>
+                  <th<% if (databaseType !== 'cassandra') { %> v-on:click="changeOrder('id')"<% } %>><span v-text="$t('global.field.id')">ID</span><% if (databaseType !== 'cassandra') { %> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'id'"></jhi-sort-indicator><% } %></th>
+                  <th<% if (databaseType !== 'cassandra') { %> v-on:click="changeOrder('login')"<% } %>><span v-text="$t('userManagement.login')">Login</span><% if (databaseType !== 'cassandra') { %> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'login'"></jhi-sort-indicator><% } %></th>
+                  <th<% if (databaseType !== 'cassandra') { %> v-on:click="changeOrder('email')"<% } %>><span v-text="$t('userManagement.email')">Email</span><% if (databaseType !== 'cassandra') { %> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'email'"></jhi-sort-indicator><% } %></th>
+                  <th></th>
+                  <%_ if (enableTranslation) { _%>
+                  <th<% if (databaseType !== 'cassandra') { %> v-on:click="changeOrder('langKey')"<% } %>> <span v-text="$t('userManagement.langKey')">Lang Key</span><% if (databaseType !== 'cassandra') { %> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'langKey'"></jhi-sort-indicator><% } %></th>
+                  <%_ } _%>
+                  <th><span v-text="$t('userManagement.profiles')">Profiles</span></th>
+                  <%_ if (databaseType !== 'cassandra') { _%>
+                  <th v-on:click="changeOrder('createdDate')"><span v-text="$t('userManagement.createdDate')">Created Date</span> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'createdDate'"></jhi-sort-indicator></th>
+                  <th v-on:click="changeOrder('lastModifiedBy')"><span v-text="$t('userManagement.lastModifiedBy')">Last Modified By</span> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'lastModifiedBy'"></jhi-sort-indicator></th>
+                  <th id="modified-date-sort" v-on:click="changeOrder('lastModifiedDate')"><span v-text="$t('userManagement.lastModifiedDate')">Last Modified Date</span> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'lastModifiedDate'"></jhi-sort-indicator></th>
+                  <%_ } _%>
+                  <th></th>
                 </tr>
                 </thead>
                 <tbody v-if ="users">

--- a/generators/client/templates/vue/src/main/webapp/app/main.ts.ejs
+++ b/generators/client/templates/vue/src/main/webapp/app/main.ts.ejs
@@ -8,6 +8,7 @@ import router from './router';
 import * as config from './shared/config/config';
 import * as bootstrapVueConfig from './shared/config/config-bootstrap-vue';
 import JhiItemCountComponent from './shared/jhi-item-count.vue';
+import JhiSortIndicatorComponent from "./shared/sort/jhi-sort-indicator.vue";
 <%_ if ((databaseType !== 'no' || authenticationType === 'uaa') && databaseType !== 'cassandra') { _%>
 import AuditsService from './admin/audits/audits.service';
 <%_ } %>
@@ -43,6 +44,7 @@ bootstrapVueConfig.initBootstrapVue(Vue);
 Vue.use(Vue2Filters)
 Vue.component('font-awesome-icon', FontAwesomeIcon);
 Vue.component('jhi-item-count', JhiItemCountComponent);
+Vue.component('jhi-sort-indicator', JhiSortIndicatorComponent);
 
 <%_ if (enableTranslation) { _%>
 const i18n = config.initI18N(Vue);

--- a/generators/client/templates/vue/src/main/webapp/app/shared/config/config.ts.ejs
+++ b/generators/client/templates/vue/src/main/webapp/app/shared/config/config.ts.ejs
@@ -6,38 +6,40 @@ import <%=jhiPrefixCapitalized%>Formatter from './formatter';
 import { setupAxiosInterceptors } from '@/shared/config/axios-interceptor';
 
 import { library } from '@fortawesome/fontawesome-svg-core';
-import { faSort } from '@fortawesome/free-solid-svg-icons/faSort';
-import { faEye } from '@fortawesome/free-solid-svg-icons/faEye';
-import { faSync } from '@fortawesome/free-solid-svg-icons/faSync';
-import { faBan } from '@fortawesome/free-solid-svg-icons/faBan';
-import { faTrash } from '@fortawesome/free-solid-svg-icons/faTrash';
 import { faArrowLeft } from '@fortawesome/free-solid-svg-icons/faArrowLeft';
-import { faSave } from '@fortawesome/free-solid-svg-icons/faSave';
-import { faPlus } from '@fortawesome/free-solid-svg-icons/faPlus';
-import { faPencilAlt } from '@fortawesome/free-solid-svg-icons/faPencilAlt';
-import { faUser } from '@fortawesome/free-solid-svg-icons/faUser';
-import { faHdd } from '@fortawesome/free-solid-svg-icons/faHdd';
-import { faTachometerAlt } from '@fortawesome/free-solid-svg-icons/faTachometerAlt';
-import { faHeart } from '@fortawesome/free-solid-svg-icons/faHeart';
-import { faList } from '@fortawesome/free-solid-svg-icons/faList';
-import { faTasks } from '@fortawesome/free-solid-svg-icons/faTasks';
+import { faAsterisk } from '@fortawesome/free-solid-svg-icons/faAsterisk';
+import { faBan } from '@fortawesome/free-solid-svg-icons/faBan';
+import { faBars } from '@fortawesome/free-solid-svg-icons/faBars';
+import { faBell } from '@fortawesome/free-solid-svg-icons/faBell';
 import { faBook } from '@fortawesome/free-solid-svg-icons/faBook';
+import { faCloud } from '@fortawesome/free-solid-svg-icons/faCloud';
+import { faEye } from '@fortawesome/free-solid-svg-icons/faEye';
+import { faFlag } from '@fortawesome/free-solid-svg-icons/faFlag';
+import { faHdd } from '@fortawesome/free-solid-svg-icons/faHdd';
+import { faHeart } from '@fortawesome/free-solid-svg-icons/faHeart';
+import { faHome } from '@fortawesome/free-solid-svg-icons/faHome';
+import { faList } from '@fortawesome/free-solid-svg-icons/faList';
 import { faLock } from '@fortawesome/free-solid-svg-icons/faLock';
+import { faPencilAlt } from '@fortawesome/free-solid-svg-icons/faPencilAlt';
+import { faPlus } from '@fortawesome/free-solid-svg-icons/faPlus';
+import { faRoad } from '@fortawesome/free-solid-svg-icons/faRoad';
+import { faSave } from '@fortawesome/free-solid-svg-icons/faSave';
+import { faSearch } from '@fortawesome/free-solid-svg-icons/faSearch';
 import { faSignInAlt } from '@fortawesome/free-solid-svg-icons/faSignInAlt';
 import { faSignOutAlt } from '@fortawesome/free-solid-svg-icons/faSignOutAlt';
+import { faSort } from '@fortawesome/free-solid-svg-icons/faSort';
+import { faSortDown } from '@fortawesome/free-solid-svg-icons/faSortDown';
+import { faSortUp } from '@fortawesome/free-solid-svg-icons/faSortUp';
+import { faSync } from '@fortawesome/free-solid-svg-icons/faSync';
+import { faTachometerAlt } from '@fortawesome/free-solid-svg-icons/faTachometerAlt';
+import { faTasks } from '@fortawesome/free-solid-svg-icons/faTasks';
 import { faThList } from '@fortawesome/free-solid-svg-icons/faThList';
-import { faUserPlus } from '@fortawesome/free-solid-svg-icons/faUserPlus';
-import { faWrench } from '@fortawesome/free-solid-svg-icons/faWrench';
-import { faAsterisk } from '@fortawesome/free-solid-svg-icons/faAsterisk';
-import { faFlag } from '@fortawesome/free-solid-svg-icons/faFlag';
-import { faBell } from '@fortawesome/free-solid-svg-icons/faBell';
-import { faHome } from '@fortawesome/free-solid-svg-icons/faHome';
 import { faTimesCircle } from '@fortawesome/free-solid-svg-icons/faTimesCircle';
-import { faSearch } from '@fortawesome/free-solid-svg-icons/faSearch';
-import { faRoad } from '@fortawesome/free-solid-svg-icons/faRoad';
-import { faCloud } from '@fortawesome/free-solid-svg-icons/faCloud';
-import { faBars } from '@fortawesome/free-solid-svg-icons/faBars';
 import { faTimes} from '@fortawesome/free-solid-svg-icons/faTimes';
+import { faTrash } from '@fortawesome/free-solid-svg-icons/faTrash';
+import { faUserPlus } from '@fortawesome/free-solid-svg-icons/faUserPlus';
+import { faUser } from '@fortawesome/free-solid-svg-icons/faUser';
+import { faWrench } from '@fortawesome/free-solid-svg-icons/faWrench';
 
 import VueCookie from 'vue-cookie';
 import Vuelidate from 'vuelidate';
@@ -66,38 +68,40 @@ export function initVueApp(vue) {
 
 export function initFortAwesome(vue) {
   library.add(
-    faSort,
-    faEye,
-    faSync,
-    faBan,
-    faTrash,
     faArrowLeft,
-    faSave,
-    faPlus,
-    faPencilAlt,
-    faUser,
-    faTachometerAlt,
-    faHeart,
-    faList,
-    faTasks,
+    faAsterisk,
+    faBan,
+    faBars,
+    faBell,
     faBook,
+    faCloud,
+    faEye,
+    faFlag,
     faHdd,
+    faHeart,
+    faHome,
+    faList,
     faLock,
+    faPencilAlt,
+    faPlus,
+    faRoad,
+    faSave,
+    faSearch,
     faSignInAlt,
     faSignOutAlt,
-    faWrench,
+    faSort,
+    faSortDown,
+    faSortUp,
+    faSync,
+    faTachometerAlt,
+    faTasks,
     faThList,
-    faUserPlus,
-    faAsterisk,
-    faFlag,
-    faBell,
-    faHome,
-    faRoad,
-    faCloud,
+    faTimes,
     faTimesCircle,
-    faSearch,
-    faBars,
-    faTimes
+    faTrash,
+    faUser,
+    faUserPlus,
+    faWrench
   );
 }
 

--- a/generators/client/templates/vue/src/main/webapp/app/shared/sort/jhi-sort-indicator.component.ts.ejs
+++ b/generators/client/templates/vue/src/main/webapp/app/shared/sort/jhi-sort-indicator.component.ts.ejs
@@ -1,0 +1,12 @@
+import Component from 'vue-class-component';
+import { Prop, Vue } from 'vue-property-decorator';
+
+@Component
+export default class JhiSortIndicatorComponent extends Vue {
+  @Prop()
+  currentOrder: string;
+  @Prop()
+  fieldName: string;
+  @Prop()
+  reverse: boolean;
+}

--- a/generators/client/templates/vue/src/main/webapp/app/shared/sort/jhi-sort-indicator.vue.ejs
+++ b/generators/client/templates/vue/src/main/webapp/app/shared/sort/jhi-sort-indicator.vue.ejs
@@ -1,0 +1,6 @@
+<template>
+    <font-awesome-icon :icon="currentOrder === fieldName ? (reverse ? 'sort-down' : 'sort-up') : 'sort'"></font-awesome-icon>
+</template>
+
+<script lang="ts" src="./jhi-sort-indicator.component.ts">
+</script>

--- a/generators/entity-client/templates/vue/src/main/webapp/app/entities/entity.vue.ejs
+++ b/generators/entity-client/templates/vue/src/main/webapp/app/entities/entity.vue.ejs
@@ -46,16 +46,16 @@
             <table class="table table-striped">
                 <thead>
                 <tr>
-                    <th<% if (pagination !== 'no') { %> v-on:click="changeOrder('id')"<% } %>><span v-text="$t('global.field.id')">ID</span><% if (pagination !== 'no') { %> <font-awesome-icon icon="sort"></font-awesome-icon><% } %></th>
+                    <th<% if (pagination !== 'no') { %> v-on:click="changeOrder('id')"<% } %>><span v-text="$t('global.field.id')">ID</span><% if (pagination !== 'no') { %> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'id'"></jhi-sort-indicator><% } %></th>
                     <%_ for (idx in fields) { _%>
-                    <th<% if (pagination !== 'no') { %> v-on:click="changeOrder('<%= fields[idx].fieldName%>')"<% } %>><span v-text="$t('<%=`${i18nKeyPrefix}.${fields[idx].fieldName}` %>')"><%= fields[idx].fieldNameHumanized %></span><% if (pagination !== 'no') { %> <font-awesome-icon icon="sort"></font-awesome-icon><% } %></th>
+                    <th<% if (pagination !== 'no') { %> v-on:click="changeOrder('<%= fields[idx].fieldName%>')"<% } %>><span v-text="$t('<%=`${i18nKeyPrefix}.${fields[idx].fieldName}` %>')"><%= fields[idx].fieldNameHumanized %></span><% if (pagination !== 'no') { %> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'<%= fields[idx].fieldName%>'"></jhi-sort-indicator><% } %></th>
                     <%_ } _%>
                     <%_ for (idx in relationships) { _%>
                         <%_ if (relationships[idx].relationshipType === 'many-to-one'
                         || (relationships[idx].relationshipType === 'one-to-one' && relationships[idx].ownerSide === true)
                         || (relationships[idx].relationshipType === 'many-to-many' && relationships[idx].ownerSide === true && pagination === 'no')) {
                             const fieldName = dto === 'no' ? "." + relationships[idx].otherEntityField : relationships[idx].otherEntityFieldCapitalized;_%>
-                    <th<% if (pagination !== 'no') { %> v-on:click="changeOrder('<%=relationships[idx].relationshipName + (fieldName)%>')"<% } %>><span v-text="$t('<%= `${i18nKeyPrefix}.${relationships[idx].relationshipName}` %>')"><%= relationships[idx].relationshipNameHumanized %></span><% if (pagination !== 'no') { %> <font-awesome-icon icon="sort"></font-awesome-icon><% } %></th>
+                    <th<% if (pagination !== 'no') { %> v-on:click="changeOrder('<%=relationships[idx].relationshipName + (fieldName)%>')"<% } %>><span v-text="$t('<%= `${i18nKeyPrefix}.${relationships[idx].relationshipName}` %>')"><%= relationships[idx].relationshipNameHumanized %></span><% if (pagination !== 'no') { %> <jhi-sort-indicator :current-order="propOrder" :reverse="reverse" :field-name="'<%=relationships[idx].relationshipName + (fieldName)%>'"></jhi-sort-indicator><% } %></th>
                         <%_ } _%>
                     <%_ } _%>
                     <th></th>

